autrainer/core/scripts/preprocess_script.py

```diff
@@ -196,39 +197,96 @@ def _process_chunk(
             self.datasets.items(), self.preprocessing.values()
         ):
             print(f" - {name}")
-            if preprocess is None or os.path.isdir(
-                os.path.join(dataset["path"], dataset["features_subdir"])
-            ):
+            # swap dataset handler with preprocessing handler
+            features_subdir = dataset["features_subdir"]
+            output_file_handler = autrainer.instantiate_shorthand(
+                config=dataset["file_handler"],
+                instance_of=AbstractFileHandler,
+            )
+            output_file_type = dataset["file_type"]
+            # set features_subdir to None (defaults to audio)
+            # so that iteration takes place over raw data
+            dataset["features_subdir"] = None
+            dataset["file_handler"] = preprocess["file_handler"]
+            dataset["file_type"] = preprocess["file_type"]
+            if preprocess is None:
                 continue
-
-            csvs = _get_csvs(dataset["path"])
-            dfs = _read_csvs(dataset["path"], csvs)
-            unique_files = _get_unique_files(dfs, dataset["index_column"])
-            _create_subdirs(
-                os.path.join(dataset["path"], dataset["features_subdir"]),
-                unique_files,
+            data = autrainer.instantiate_shorthand(dataset)
```

Comment:
> At a later point if the user wants to use preprocessed data with the full dataset (another config), preprocessing will not run again and there will be missing data.

This is also something this PR changes, I forgot to write it, but it's something I wanted to change anyway. Now the data preprocessing runs again, but omits *single instances* that have been extracted in the past.

The original intent of this change is to account for cases where the `preprocessing` crashes or is stopped prematurely, and the user does not to re-extract all features. But I think this will cover your concern as well?

Reply:
Ok got it, yeah this would solve the issue. But then, preprocessing has to be run explicitly for ALL datasets used (we need to keep this in mind). Right now if datasets subsets would be used we process them once for all and any dataset configuration relying on already extracted features would not need to be re-run. I think this was not clear to the user anyways and repeated execution of `autrainer preprocess` would just have been a no-op anyways.

Reply:
Added a warning in the docs in 6bf34c5ec3f79b5b1e2cbdc12d64963d76facd2e